apps/www/registry/default/ui/timeline.tsx

```diff
@@ -0,0 +1,207 @@
+import React from "react"
+import { VariantProps, cva } from "class-variance-authority"
+import { Check, Circle, X } from "lucide-react"
+
+import { cn } from "@/lib/utils"
+
+const timelineVariants = cva("grid", {
+  variants: {
+    positions: {
+      left: "[&>li]:grid-cols-[0_min-content_1fr]",
+      right: "[&>li]:grid-cols-[1fr_min-content]",
+      center: "[&>li]:grid-cols-[1fr_min-content_1fr]",
+    },
+  },
+  defaultVariants: {
+    positions: "left",
+  },
+})
+
+interface TimelineProps
+  extends React.HTMLAttributes<HTMLUListElement>,
+    VariantProps<typeof timelineVariants> {}
+
+const Timeline = React.forwardRef<HTMLUListElement, TimelineProps>(
+  ({ children, className, positions, ...props }, ref) => {
+    return (
+      <ul
+        className={cn(timelineVariants({ positions }), className)}
+        ref={ref}
+        {...props}
+      >
+        {children}
+      </ul>
+    )
+  }
+)
+Timeline.displayName = "Timeline"
+
+const timelineItemVariants = cva("grid items-center gap-x-2", {
+  variants: {
+    status: {
+      done: "text-primary",
+      default: "text-muted-foreground",
+    },
+  },
+  defaultVariants: {
+    status: "default",
+  },
+})
+
+interface TimelineItemProps
+  extends React.HTMLAttributes<HTMLLIElement>,
+    VariantProps<typeof timelineItemVariants> {}
+
+const TimelineItem = React.forwardRef<HTMLLIElement, TimelineItemProps>(
+  ({ className, status, ...props }, ref) => (
+    <li
+      className={cn(timelineItemVariants({ status }), className)}
+      ref={ref}
+      {...props}
+    />
+  )
+)
+TimelineItem.displayName = "TimelineItem"
+
+const timelineDotVariants = cva(
+  "col-start-2 col-end-3 row-start-1 row-end-1 flex size-4 items-center justify-center rounded-full border border-current",
+  {
+    variants: {
+      status: {
+        default: "[&>*]:hidden",
+        current:
+          "[&>*:not(.lucide-circle)]:hidden [&>.lucide-circle]:fill-current [&>.lucide-circle]:text-current",
+        done: "bg-primary [&>*:not(.lucide-check)]:hidden [&>.lucide-check]:text-background",
+        error:
+          "border-destructive bg-destructive [&>*:not(.lucide-x)]:hidden [&>.lucide-x]:text-background",
```

Comment:
These `.lucide-*` selectors may be less than ideal for many developers using shadcn/ui.

They may be pulling their icons from different packages, such as react-icons.